flytekit/sdk/tasks.py

```diff
@@ -488,6 +490,45 @@ def wrapper(fn):
         return wrapper
 
 
+def generic_spark_task(
```

Comment:
I dont think we need this method right?
We can directly use - SdkGenericSparkTask?

Unless you are defaulting somethings. (i see timeout)

Reply:
Yeah, we are defaulting. In-addition, I also want users to be able to look-up all task_types supported from a single place.